### ISSUE TITLE
Future covariates for exponential smoothing

### ISSUE THREAD

**Issue by hrzn** [unit8co/darts#640]

**Comment by Beerstabr** [comment]
I looked in to this for a bit. I came across two approaches.

(1) sophisticated but hard
https://www.monash.edu/business/econometrics-and-business-statistics/research/publications/ebs/wp02-15.pdf

(2) unsophisticated but easy
Doing OLS with the regressors on the target series and then performing ETS on the residuals just like AutoARIMA does as Stefan Kolassa points out in this post:
https://stats.stackexchange.com/questions/220830/holt-winters-with-exogenous-regressors-in-r

**Comment by hrzn** [comment]
Hi @Beerstabr and thanks for looking into this! We are talking about `ExponentialSmoothing` and not `StatsForecastETS` here, right?
I think we could perhaps start with the "easy" approach (which sounds very reasonable already), and potentially add the sophisticated approach later on with an option at model creation time for deciding which version to use.
Would you be willing to give a try to the easy option? It's quite possible that you could benefit from simply using a `LinearRegressionModel` for the OLS part. Basically you could fit one with `lags_future_covariates=[0]` to predict the target based on the "current" future covariate lag. You could then call this inner model with the future covariates, which would give you the OLS-forecast in the form of a `TimeSeries`, which you can simply subtract from the target to get the ETS target.

**Comment by Beerstabr** [comment]
@hrzn Yes, I'd definitely be interested in giving this a try. I'll take the approach that you outlined above.

I thought about doing it for StatsForecastETS because it is faster and automatically finds the 'best' model. Would you rather have me do it for ExponentialSmoothing instead? What would be the benefit of that? I could also do it for both.

**Comment by hrzn** [comment]
@Beerstabr You're right it is of interest for both, so feel free to do StatsForecastETS only, or both :)
Thanks!
